File: server/src/main/kotlin/dev/chargedbyte/reaktor_summer_2022/model/Hand.kt
package dev.chargedbyte.reaktor_summer_2022.model

enum class Hand {
    ROCK, PAPER, SCISSORS;

    fun beats(other: Hand): Boolean {
        return when (this) {
            ROCK -> other == SCISSORS
            PAPER -> other == ROCK
            SCISSORS -> other == PAPER
        }
    }
}

File: client/model/Hand.ts
enum Hand {
  Rock = "ROCK",
  Paper = "PAPER",
  Scissors = "SCISSORS",
}

const handBeatsHand = (self: Hand, other: Hand) => {
  switch (self) {
    case Hand.Rock:
      return other === Hand.Scissors;
    case Hand.Paper:
      return other === Hand.Rock;
    case Hand.Scissors:
      return other === Hand.Paper;
  }
};

export default Hand;

export { handBeatsHand };
